src/_pytest/_code/code.py

```diff
@@ -591,7 +591,7 @@ def getrepr(
         )
         return fmt.repr_excinfo(self)
 
-    def match(self, regexp: Union[str, Pattern]) -> bool:
+    def match(self, regexp: Union[str, "Pattern"]) -> bool:
```

Comment:
I suggest we move this logic to `_pytest.compat`:

```python
from typing import Pattern

if sys.version_info <= (3, 5, 1):
    # see #5751
    PatternT = "Pattern"
else:
    PatternT = Pattern
```

And use `PatternT` where it is needed. This way the "hack" is explicit about which version it is required and easier to replace by the standard pattern later.